.github/workflows/ci.yml

```diff
@@ -0,0 +1,80 @@
+name: CI
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+env:
+  app: sidekiq_publisher
+
+jobs:
+  lint:
+    name: Lint (Ruby ${{ matrix.ruby }})
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [ '3.0', '3.1', '3.2' ]
+
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v3
+
+      - name: Install Ruby ${{ matrix.ruby }}
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+
+      - name: Install dependencies
+        run: bundle install
```

Comment:
I think you can remove this since `bundler-cache: true` will automatically install the dependencies.